src/amo/reducers/viewContext.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  VIEW_CONTEXT_EXPLORE,
+  VIEW_CONTEXT_HOME,
+  SET_VIEW_CONTEXT,
+} from 'core/constants';
+
+
+export type ViewContextType = {|
+  context: ADDON_TYPE_EXTENSION | ADDON_TYPE_THEME | VIEW_CONTEXT_EXPLORE |
```

Comment:
Cool, that worked. But now even if I change everything to what @rpl suggested it doesn't complain if I say "context can only be a number" or I submit a bad value.